### PR TITLE
reduces package size

### PR DESCRIPTION
**Pull request by vitkon**
#### What's this PR do?
- reduces the package size from 556KB to 6KB
- removes lodash (couldn't reduce package size with it less than 104KB)
- adds minification and gzip
- adds peer dependencies

#### Where should the reviewer start?
package.json, webpack.config.js

#### How should this be manually tested?
compare build sizes on master and this branch

#### Any background context you want to provide?
- lodash util functions were substituted with custom ones hence they need testing added.

#### What are the relevant tickets / issues?
#2 

#### Screenshots (if appropriate)
screen shot 2018-03-25 at 23 40 49

### CODE REVIEW

File: src/validate.ts [hunk 1]
```diff
@@ -34,12 +35,15 @@ const getValidationResult = ({
 const inferRulesFromAttributes = (rules: any[], { inputs }: any) => {
     const extendedRules = [...rules];
 
-    forEach(inputs, (input: any) => {
-        if (get(input, 'validity.valid') === false) {
-            const rule = hasError(input.name, input.validationMessage);
-            extendedRules.push(rule);
+    for (const prop in inputs) {
+        if (inputs.hasOwnProperty(prop)) {
```
Comment (mattdell) [inline src/validate.ts]: I know eslint discourages this: https://eslint.org/docs/rules/no-prototype-builtins

Safe to use?

Comment (vitkon) [reply src/validate.ts]: good point, but as far as I understand we don't use `Object.create(null)` for inputs

Comment (mattdell) [reply src/validate.ts]: Cool, one to keep in mind I suppose if we get any issues.

**Comment by mattdell** [comment]
Looks great!

**Pull request merged**